pkg/retrieval/retrieval.go

```diff
@@ -126,7 +126,7 @@ func (s *Service) RetrieveChunk(ctx context.Context, addr swarm.Address, origin
 		var (
 			peerAttempt  int
 			peersResults int
-			resultC      = make(chan retrievalResult, maxSelects)
```

Comment:
why is a buffered channel larger than 1 needed here? the separate goroutines can just try to write to the channel with a select-default block, and then it is not needed. i find the current implementation a bit convoluted, maybe we could simplify it a bit? not sure why it is needed for example to communicate an empty result. these patterns can be simplified using waitgroups